src/lib.rs

```diff
@@ -158,7 +158,6 @@ pub(crate) mod allocator;
 #[cfg(feature = "visualizer")]
 pub mod visualizer;
 
-#[cfg(all(not(any(target_os = "macos", target_os = "ios")), feature = "vulkan"))]
```

Comment:
You should still leave the `vulkan` feature in place:

```suggestion
#[cfg(feature = "vulkan")]
```

Other than that I agree that `ash` with `ash-molten` should work just fine on Apple products.